pkg/internal/layers/layers.go

```diff
@@ -85,11 +69,33 @@ type Layer interface {
 	GetAddonsLayers() (map[string]*kraanv1alpha1.AddonsLayer, error)
 }
 
+// KraanLayer is the Schema for the addons API.
+type KraanLayer struct {
```

Comment:
Just for understanding, so we have a `KraanLayer` which appears to be a canonical representation of a layer and then we have `kraanv1alpha1.AddonsLayer` which is the CRD definition. So this is done to isolate the internal logic from CRD version changes?